site/content/blog/2020-03-16-advanced-git-tbilisi-free-university.md

```diff
@@ -0,0 +1,437 @@
+4fe98edfa HEAD@{10}: rebase -i (squash): # This is a combination of 2 commits.
+da0d04f00 HEAD@{11}: rebase -i (start): checkout origin/master
+3489625af HEAD@{12}: rebase -i (finish): returning to refs/heads/test-git-rebase
+3489625af HEAD@{13}: rebase -i (pick): wip
+f1ef494d4 HEAD@{14}: rebase -i (pick): remove dead code
+b57fde469 HEAD@{15}: rebase -i (pick): improve testing coe
+da0d04f00 HEAD@{16}: rebase -i (pick): add TestConfigEnableDeveloper
+1c498996a (origin/master, origin/HEAD, master) HEAD@{17}: rebase -i (start): checkout origin/master
+```
+
+You can see the rebase we just did, and the rebase before that, and the rebase before that. If I run the command again with another flag, I can see the time when these references changed:
+
+```sh
+git reflog --date=iso
+```
+
+```
+bd8501963 (HEAD -> test-git-rebase) HEAD@{2020-03-16 18:01:49 -0300}: rebase -i (finish): returning to refs/heads/test-git-rebase
+bd8501963 (HEAD -> test-git-rebase) HEAD@{2020-03-16 18:01:49 -0300}: rebase -i (start): checkout origin/master
+353db22f6 HEAD@{2020-03-16 18:01:45 -0300}: rebase -i (finish): returning to refs/heads/test-git-rebase
+353db22f6 HEAD@{2020-03-16 18:01:45 -0300}: rebase -i (pick): remove dead code
+bd8501963 (HEAD -> test-git-rebase) HEAD@{2020-03-16 18:01:43 -0300}: rebase -i (reword): add TestConfigEnableDeveloper
+e9d6796e8 HEAD@{2020-03-16 18:01:43 -0300}: rebase -i: fast-forward
+1c498996a (origin/master, origin/HEAD, master) HEAD@{2020-03-16 18:01:43 -0300}: rebase -i (start): checkout origin/master
+9ee8564a3 HEAD@{2020-03-16 18:01:37 -0300}: rebase -i (finish): returning to refs/heads/test-git-rebase
+9ee8564a3 HEAD@{2020-03-16 18:01:37 -0300}: rebase -i (pick): remove dead code
+e9d6796e8 HEAD@{2020-03-16 18:01:35 -0300}: rebase -i (squash): add TestConfigEnableDeveloper
+4fe98edfa HEAD@{2020-03-16 18:01:35 -0300}: rebase -i (squash): # This is a combination of 2 commits.
+da0d04f00 HEAD@{2020-03-16 18:01:35 -0300}: rebase -i (start): checkout origin/master
+3489625af HEAD@{2020-03-16 17:53:19 -0300}: rebase -i (finish): returning to refs/heads/test-git-rebase
+3489625af HEAD@{2020-03-16 17:53:19 -0300}: rebase -i (pick): wip
+f1ef494d4 HEAD@{2020-03-16 17:53:19 -0300}: rebase -i (pick): remove dead code
+b57fde469 HEAD@{2020-03-16 17:53:19 -0300}: rebase -i (pick): improve testing coe
+da0d04f00 HEAD@{2020-03-16 17:53:19 -0300}: rebase -i (pick): add TestConfigEnableDeveloper
+1c498996a (origin/master, origin/HEAD, master) HEAD@{2020-03-16 17:53:19 -0300}: rebase -i (start): checkout origin/master
+```
+
+And there's my missing commit reference! I can `git show f1ef494d4` it to confirm, and then bring it back into my branch with a simple `git cherry-pick f1ef494d4`.
+
+I presented `git reflog` last, since I think it's one of the harder commands to fundamentally understand: but this is probably the one command you should learn first, because you recover from almost any mistake you make with git using the reflog.
```

Comment:
```suggestion
I presented `git reflog` last, since I think it's one of the harder commands to fundamentally understand: but this is probably the one command you should learn first, because you recover from almost any mistake you make with Git using the reflog.
```